opm/simulators/flow/equil/EquilibrationHelpers_impl.hpp

```diff
@@ -448,6 +448,13 @@ operator()(double s) const
     fluidState.setSaturation(FluidSystem::gasPhaseIdx, 0.0);
     fluidState.setSaturation(phase_, s);
 
+    // This code should only be called for water or gas phase
+    assert(phase_ != FluidSystem::oilPhaseIdx);
+    // The capillaryPressure code only uses the wetting phase saturation
+    if (phase_ == FluidSystem::gasPhaseIdx) {
+        fluidState.setSaturation(FluidSystem::waterPhaseIdx, 1.0 - s);
```

Comment:
Should this not be 0? Or is this a small trick to handle both gas-water and gas-oil systems?

Reply:
Yes. It the later code will only use the wetting phase saturation. So water for gas-water and oil-water systems and oil for oil-gas systems.